src/util/bitvector_expr.h

```diff
@@ -735,6 +735,9 @@ class binary_overflow_exprt : public binary_predicate_exprt
   {
     check(expr, vm);
   }
+
+protected:
+  using binary_predicate_exprt::binary_predicate_exprt;
```

Comment:
Why did this turn out to be necessary?

Reply:
Strictly speaking it isn't. I could have reused the existing constructor of `binary_overflow_exprt`, in the constructors of the new sub-classes. This would have required specifying the kind as `"plus"` rather than `ID_overflow_plus` for example. So the constructor would have been converting c-string -> `irep_idt` -> `std::string` -> string concatenation -> `irep_idt`. The way I have implemented this uses the end result `irep_idt` directly, which saves the aforementioned conversions and makes it easier for readers of the sub-class to work out what the constructor sets the id to.

Note that constructors of sub-classes need to be able to explicitly construct their base classes if no default constructor is available and that even public constructors are not available for the construction of derived classes by default. Adding `using binary_predicate_exprt::binary_predicate_exprt;` allows me to use the constructor of `binary_predicate_exprt` to construct `binary_overflow_exprt`, without writing a whole constructor just to forward its arguments onto the base constructor.

Reply:
The approach I outlined here doesn't actually work, due to the constructor of `binary_overflow_exprt` hiding the constructor of `binary_predicate_exprt`. This has been bug fixed in the separated PR - https://github.com/diffblue/cbmc/pull/6744